### DISCUSSION TITLE
Recording clips in H.264 rather than HEVC

### DISCUSSION THREAD

**Discussion by wrightbuilt** [blakeblackshear/frigate#8143] [Ask A Question]
Is there a way to have my clips record in H.264 rather than HEVC? I don't have much experience with ffmpeg so I am not sure how to approach this.

**Comment by NickM-27** [comment]
What type of camera? most would configure this in the camera itself. 

Otherwise can use go2rtc

**Reply by wrightbuilt** [reply]
It is a Reolink Duo 2. I am specifying the H264 feed for the camera. But the detect objects clips are in HEVC. I specifically want the clips to be in H.264

**Reply by NickM-27** [reply]
Yeah reolink doesn't offer that. What is your config?

**Reply by wrightbuilt** [reply]
I don't know what you mean, doesn't offer that. The streams I am using are H264.

mqtt:
  enabled: True
  user: "***"
  password: "***"
  host: ***
  port: 1883
  
detectors:
  coral:
    type: edgetpu
    device: usb

cameras:
  Front:
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://frigate:***@192.168.110.206:554/h264Preview_01_main 
          roles:
            - record
        - path: rtsp://frigate:***@192.168.110.206:554/h264Preview_01_sub
          roles:
            - detect
    zones:
      porch:
        coordinates: 212,105,0,287,0,480,640,480,640,0,387,0,381,104
    detect:
      enabled: True 
      width: 640 
      height: 480 
    snapshots:
      required_zones:
        - porch
      enabled: True
    record:
      events:
        retain:
          default: 14
          mode: active_objects
        required_zones:
          - porch
      enabled: True

  Garage:
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://frigate:***@192.168.110.205:554/h264Preview_01_main 
          roles:
            - record
        - path: rtsp://frigate:***@192.168.110.205:554/h264Preview_01_sub
          roles:
            - detect
    zones:
      main_garage:
        coordinates: 0,480,640,480,640,0,0,0
        objects:
          - person
          - cat
          - dog
          - bicycle
          - package
    detect:
      enabled: True 
      width: 640 
      height: 480 
    snapshots:
      enabled: True
      required_zones:
        - main_garage
    record:
      events:
        retain:
          default: 14
          mode: active_objects
        required_zones:
          - main_garage
      enabled: True

  Back:
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://frigate:***@192.168.110.207:554/h264Preview_01_main 
          roles:
            - record
        - path: rtsp://frigate:***@192.168.110.207:554/h264Preview_01_sub
          roles:
            - detect
    zones:
      patio:
        coordinates: 141,114,0,257,0,480,640,480,640,0,404,0,394,46
        objects:
          - person
    detect:
      enabled: True 
      width: 640 
      height: 480 
    snapshots:
      required_zones:
        - patio
      enabled: True
    record:
      events:
        retain:
          default: 14
          mode: active_objects
        required_zones:
          - patio
      enabled: True

  Driveway:
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://frigate:***@192.168.110.208:554/h264Preview_01_main 
          roles:
            - record
        - path: rtsp://frigate:***@192.168.110.208:554/h264Preview_01_sub
          roles:
            - detect
    zones:
      main_driveway:
        inertia: 3
        coordinates: 35,463,0,440,0,576,1536,576,1536,284,1397,435,1195,355,949,318,519,330,479,206,273,388
        filters:
          person:
            min_area: 2000
      backyard:
        inertia: 3
        coordinates: 35,463,273,388,483,188,125,270,0,326,0,443
        objects:
          - person
        filters:
          person:
            min_area: 2000    
      alley:
        coordinates: 1280,0,1536,284,1397,435,1195,355,949,318,519,330,479,206,123,270,0,326,0,207,190,0
        objects:
          - person
          - cat
          - dog
          - bicycle
        filters:
          person:
            min_area: 2000 
    motion:
      mask:
        - 1022,68,1015,123,852,88,855,30
    detect:
      enabled: True 
      width: 1536 
      height: 576 
    snapshots:
      required_zones:
        - main_driveway
        - backyard
        - alley
      enabled: True
    record:
      events:
        retain:
          default: 14
          mode: active_objects
        required_zones:
          - main_driveway
          - backyard
          - alley
      enabled: True

objects:
  track:
    - car
    - motorcycle
    - bus
    - person
    - cat
    - dog
    - bicycle
    - package

birdseye:
  enabled: True

**Reply by NickM-27** [reply]
The reolink duo 2 does not support h264 for the main stream. It does not matter if you put h264 in the url. You can check with vlc or ffprobe, the main stream is h265 only (which is one of the main issues with reolink because they don't give the option to change it)

**Reply by wrightbuilt** [reply]
awesome

**Reply by NickM-27** [reply]
So the only option for h264 would be to configure go2rtc to transcode the stream.

**Reply by wrightbuilt** [reply]
Thanks I will give that a shot in the morning
